android/src/main/java/com/rogchap/react/modules/twilio/TwilioModule.java

```diff
@@ -146,7 +146,19 @@ public void onError(Exception e) {
                 }
             });
         } else {
-            sendEvent("deviceReady", null);
+            try {
+                if (_phone != null) {
+                    _phone.release();
+                }
+                _phone = Twilio.createDevice(token, dl);
```

Comment:
@sagivmalihi I understand your goal, but there is a lot of copy and paste here. Would you mind if I refactor this to be more DRY.
Out of curiosity did you need to ad this code to prevent a crash or something bad? Because I have a crash that I am struggling to fix